### PR TITLE
jekyll: enable sitemap and local search for dev environment

### PR DESCRIPTION
**Pull request by crazy-max**
closes https://github.com/docker/docker.github.io/pull/14784
also related to https://github.com/docker/docker.github.io/pull/15179 as well as follow-up https://github.com/docker/docker.github.io/pull/15153

To test changes for local search but also the sitemap in our lab environment we need to enable these features. Since Jekyll 4, sitemap generation as well as metadata.json is pretty well optimized and takes less than 1 second for our docs so I think we can enable them for all environments which will allow us to check if nothing is broken on netlify environments before merging PRs.

Stats for sitemap and local search disabled:

```
#14 34.53 Build Process Summary:
#14 34.53 
#14 34.53 | PHASE      |    TIME |
#14 34.53 +------------+---------+
#14 34.53 | RESET      |  0.0004 |
#14 34.53 | READ       | 19.5737 |
#14 34.53 | GENERATE   |  1.1804 |
#14 34.53 | RENDER     | 12.7393 |
#14 34.53 | CLEANUP    |  0.0185 |
#14 34.53 | WRITE      |  0.2298 |
#14 34.53 +------------+---------+
#14 34.53 | TOTAL TIME | 33.7421 |
#14 34.53
#14 34.53
#14 34.53 Site Render Stats:
#14 34.54
#14 34.54 | Filename                                           | Count |     Bytes |   Time |
#14 34.54 +----------------------------------------------------+-------+-----------+--------+
#14 34.54 | _layouts/docs.html                                 |   712 | 23595.86K |  6.711 |
#14 34.54 | _includes/header.html                              |   712 |  1780.88K |  4.597 |
#14 34.54 | _includes/breadcrumbs.html                         |   712 |   301.95K |  4.532 |
#14 34.54 | _includes/toc_pure_liquid.html                     |   699 |   566.29K |  1.466 |
#14 34.54 | _includes/cli.md                                   |   243 |  1488.15K |  0.266 |
#14 34.54 | _includes/read_time.html                           |   712 |    21.14K |  0.175 |
#14 34.54 | _includes/head.html                                |   713 |  3098.72K |  0.172 |
#14 34.54 | _includes/navigation.html                          |   712 |  1179.25K |  0.045 |
#14 34.54 | _layouts/redirect.html                             |  1059 |   484.21K |  0.043 |
#14 34.54 | sitemap.xml                                        |     1 |     1.12K |  0.029 |
#14 34.54 | glossary.md                                        |     1 |    30.24K |  0.026 |
#14 34.54 | get-started/orchestration.md                       |     1 |    19.13K |  0.010 |
#14 34.54 | engine/reference/commandline/run.md                |     1 |    51.60K |  0.010 |
#14 34.54 | engine/reference/commandline/service_update.md     |     1 |    28.39K |  0.010 |
#14 34.54 | engine/reference/commandline/container_run.md      |     1 |    24.12K |  0.009 |
#14 34.54 | engine/reference/commandline/service_create.md     |     1 |    58.24K |  0.009 |
#14 34.54 | engine/reference/commandline/create.md             |     1 |    21.48K |  0.009 |
#14 34.54 | engine/reference/commandline/container_create.md   |     1 |    23.56K |  0.008 |
#14 34.54 | _includes/footer.html                              |   713 |  4949.22K |  0.006 |
#14 34.54 | engine/reference/commandline/buildx_build.md       |     1 |    27.54K |  0.005 |
#14 34.54 | _includes/theme-switch.html                        |   713 |   264.59K |  0.004 |
#14 34.54 | engine/reference/commandline/build.md              |     1 |    35.58K |  0.004 |
#14 34.54 | engine/reference/commandline/image_build.md        |     1 |    10.08K |  0.004 |
#14 34.54 | engine/reference/commandline/builder_build.md      |     1 |     8.30K |  0.003 |
#14 34.54 | engine/reference/commandline/app.md                |     1 |    17.30K |  0.003 |
#14 34.54 | _includes/upgrade-cta.html                         |    11 |     6.76K |  0.003 |
#14 34.54 | engine/reference/commandline/compose_up.md         |     1 |    10.40K |  0.003 |
#14 34.54 | engine/reference/commandline/buildx_bake.md        |     1 |    26.05K |  0.002 |
#14 34.54 | engine/reference/commandline/container_update.md   |     1 |     8.61K |  0.002 |
#14 34.54 | engine/reference/commandline/compose_run.md        |     1 |    10.68K |  0.002 |
#14 34.54 | engine/reference/commandline/compose_build.md      |     1 |     8.06K |  0.002 |
#14 34.54 | engine/reference/commandline/swarm_init.md         |     1 |    12.24K |  0.002 |
#14 34.54 | engine/reference/commandline/network_create.md     |     1 |    13.36K |  0.002 |
#14 34.54 | engine/reference/commandline/buildx_create.md      |     1 |    11.87K |  0.002 |
#14 34.54 | engine/reference/commandline/compose.md            |     1 |    11.42K |  0.002 |
#14 34.54 | engine/reference/commandline/compose_convert.md    |     1 |     7.78K |  0.002 |
#14 34.54 | engine/reference/commandline/container_exec.md     |     1 |     7.35K |  0.002 |
#14 34.54 | engine/reference/commandline/container_ls.md       |     1 |     7.12K |  0.002 |
#14 34.54 | engine/reference/commandline/update.md             |     1 |     6.73K |  0.002 |
#14 34.54 | engine/reference/commandline/plugin_push.md        |     1 |     3.74K |  0.002 |
#14 34.54 | engine/reference/commandline/docker.md             |     1 |     9.74K |  0.002 |
#14 34.54 | js/toc.json                                        |     1 |    56.86K |  0.002 |
#14 34.54 | engine/reference/commandline/ps.md                 |     1 |    21.69K |  0.002 |
#14 34.54 | engine/reference/commandline/compose_exec.md       |     1 |     7.46K |  0.002 |
#14 34.54 | engine/reference/commandline/compose_ps.md         |     1 |    10.43K |  0.002 |
#14 34.54 | engine/reference/commandline/service_logs.md       |     1 |     6.92K |  0.002 |
#14 34.54 | engine/reference/commandline/container_logs.md     |     1 |     6.85K |  0.002 |
#14 34.54 | desktop/previous-versions/edge-releases-windows.md |     1 |   194.28K |  0.002 |
#14 34.54 | engine/reference/commandline/compose_logs.md       |     1 |     6.81K |  0.002 |
#14 34.54 | engine/reference/commandline/stack_ps.md           |     1 |    15.59K |  0.002 |
#14 34.54 +----------------------------------------------------+-------+-----------+--------+
#14 34.54 | TOTAL (for 50 files)                               |  7749 | 38575.77K | 18.202 |
#14 34.54
#14 34.54                     done in 33.755 seconds.
#14 34.54  Auto-regeneration: disabled. Use --watch to enable.
#14 34.56 + mkdir -p /out/js
#14 34.57 + echo '[]'
#14 DONE 36.3s
```

Enabled:

```
#12 32.94 Build Process Summary:
#12 32.94
#12 32.94 | PHASE      |    TIME |
#12 32.94 +------------+---------+
#12 32.94 | RESET      |  0.0002 |
#12 32.94 | READ       | 17.7280 |
#12 32.94 | GENERATE   |  1.2210 |
#12 32.94 | RENDER     | 13.0898 |
#12 32.94 | CLEANUP    |  0.0141 |
#12 32.94 | WRITE      |  0.2356 |
#12 32.94 +------------+---------+
#12 32.94 | TOTAL TIME | 32.2887 |
#12 32.94
#12 32.94
#12 32.94 Site Render Stats:
#12 32.95
#12 32.95 | Filename                                         | Count |     Bytes |   Time |
#12 32.95 +--------------------------------------------------+-------+-----------+--------+
#12 32.95 | _layouts/docs.html                               |   712 | 23625.76K |  6.742 |
#12 32.95 | _includes/header.html                            |   712 |  1780.88K |  4.585 |
#12 32.95 | _includes/breadcrumbs.html                       |   712 |   301.95K |  4.519 |
#12 32.95 | _includes/toc_pure_liquid.html                   |   699 |   566.29K |  1.487 |
#12 32.95 | _includes/cli.md                                 |   243 |  1488.15K |  0.271 |
#12 32.95 | _includes/read_time.html                         |   712 |    21.14K |  0.182 |
#12 32.95 | _includes/head.html                              |   713 |  3128.66K |  0.180 |
#12 32.95 | js/metadata.json                                 |     1 |   146.05K |  0.145 |
#12 32.95 | sitemap.xml                                      |     1 |    42.55K |  0.082 |
#12 32.95 | _includes/navigation.html                        |   712 |  1179.25K |  0.046 |
#12 32.95 | _layouts/redirect.html                           |  1059 |   484.21K |  0.040 |
#12 32.95 | glossary.md                                      |     1 |    30.24K |  0.025 |
#12 32.95 | engine/reference/commandline/service_update.md   |     1 |    28.39K |  0.010 |
#12 32.95 | get-started/orchestration.md                     |     1 |    19.13K |  0.010 |
#12 32.95 | engine/reference/commandline/container_create.md |     1 |    23.56K |  0.009 |
#12 32.95 | engine/reference/commandline/run.md              |     1 |    51.60K |  0.009 |
#12 32.95 | engine/reference/commandline/container_run.md    |     1 |    24.12K |  0.009 |
#12 32.95 | engine/reference/commandline/service_create.md   |     1 |    58.24K |  0.008 |
#12 32.95 | engine/reference/commandline/create.md           |     1 |    21.48K |  0.008 |
#12 32.95 | _includes/footer.html                            |   713 |  4949.22K |  0.007 |
#12 32.95 | engine/reference/commandline/buildx_build.md     |     1 |    27.54K |  0.005 |
#12 32.95 | engine/reference/commandline/build.md            |     1 |    35.58K |  0.005 |
#12 32.95 | _includes/theme-switch.html                      |   713 |   264.59K |  0.004 |
#12 32.95 | engine/reference/commandline/app.md              |     1 |    17.30K |  0.004 |
#12 32.95 | engine/reference/commandline/builder_build.md    |     1 |     8.30K |  0.004 |
#12 32.95 | engine/reference/commandline/image_build.md      |     1 |    10.08K |  0.003 |
#12 32.95 | _includes/upgrade-cta.html                       |    11 |     6.76K |  0.003 |
#12 32.95 | engine/reference/commandline/compose_up.md       |     1 |    10.40K |  0.003 |
#12 32.95 | engine/reference/commandline/compose_run.md      |     1 |    10.68K |  0.003 |
#12 32.95 | engine/reference/commandline/buildx_bake.md      |     1 |    26.05K |  0.002 |
#12 32.95 | engine/reference/commandline/swarm_init.md       |     1 |    12.24K |  0.002 |
#12 32.95 | engine/reference/commandline/compose_build.md    |     1 |     8.06K |  0.002 |
#12 32.95 | engine/reference/commandline/update.md           |     1 |     6.73K |  0.002 |
#12 32.95 | engine/reference/commandline/container_update.md |     1 |     8.61K |  0.002 |
#12 32.95 | engine/reference/commandline/compose.md          |     1 |    11.42K |  0.002 |
#12 32.95 | engine/reference/commandline/buildx_create.md    |     1 |    11.87K |  0.002 |
#12 32.95 | engine/reference/commandline/network_create.md   |     1 |    13.36K |  0.002 |
#12 32.95 | engine/reference/commandline/app_validate.md     |     1 |     4.28K |  0.002 |
#12 32.95 | engine/reference/commandline/compose_convert.md  |     1 |     7.78K |  0.002 |
#12 32.95 | engine/reference/commandline/container_exec.md   |     1 |     7.35K |  0.002 |
#12 32.95 | engine/reference/commandline/compose_exec.md     |     1 |     7.46K |  0.002 |
#12 32.95 | engine/reference/commandline/app_install.md      |     1 |     6.53K |  0.002 |
#12 32.95 | engine/reference/commandline/compose_ps.md       |     1 |    10.43K |  0.002 |
#12 32.95 | engine/reference/commandline/docker.md           |     1 |     9.74K |  0.002 |
#12 32.95 | engine/reference/commandline/container_ls.md     |     1 |     7.12K |  0.002 |
#12 32.95 | engine/reference/commandline/ps.md               |     1 |    21.69K |  0.002 |
#12 32.95 | engine/reference/commandline/compose_pull.md     |     1 |     8.40K |  0.002 |
#12 32.95 | engine/reference/commandline/stack_ps.md         |     1 |    15.59K |  0.002 |
#12 32.95 | desktop/previous-versions/edge-releases-mac.md   |     1 |   184.66K |  0.002 |
#12 32.95 | js/toc.json                                      |     1 |    56.86K |  0.002 |
#12 32.95 +--------------------------------------------------+-------+-----------+--------+
#12 32.95 | TOTAL (for 50 files)                             |  7749 | 38808.36K | 18.446 |
#12 32.95
#12 32.95                     done in 32.303 seconds.
#12 32.95  Auto-regeneration: disabled. Use --watch to enable.
#12 DONE 34.7s
```

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -44,27 +44,18 @@ COPY --from=vendored /out /
...
-  )
-fi
+CONFIG_FILES=_config.yml$([ "$JEKYLL_ENV" = "production" ] && echo ",_config_production.yml" || true)
+(
+  set -x
+  bundle exec jekyll build --profile -d ${TARGET} --config ${CONFIG_FILES}
+)
 find ${TARGET} -type f -name '*.html' | while read i; do
-  sed -i 's#\(<a[^>]* href="\)https://${DOMAIN}/#\1/#g' "$i"
+  sed -i 's#\(<a[^>]* href="\)${DOCS_URL}/#\1/#g' "$i"
```
Comment (crazy-max) [inline Dockerfile:58]: @thaJeztah About what you said in https://github.com/docker/docker.github.io/pull/14784#discussion_r878382327, do we want to tackle this here or in a follow-up?

**Comment by crazy-max** [comment]
> https://deploy-preview-15250--docsdocker.netlify.app/sitemap.xml

image

File: Dockerfile (hunk 1 above)
Comment (thaJeztah) [reply Dockerfile:58]: Probably best to not change it (for now) until we have a full understanding of consequences (if we _would_ change)?

Comment (crazy-max) [reply Dockerfile:58]: `DOCS_URL` is needed for `update_sitemap.rb` plugin but here it doesn't change the current behavior on prod and will still be `https://docs.docker.com`

**Comment by thaJeztah** [comment]
Hmm... looks like something is broken in general on docs.docker.com; https://docs.docker.com/sitemap.xml looks like the env-var is not being substituted; did we break that somewhere along the line?

Funny thing is that the _first entry_ actually looks to have a date set, but items following it don't?

Screenshot 2022-07-28 at 14 11 24

**Comment by crazy-max** [comment]
@thaJeztah Rebased and also moved the URLs fix script from the Dockerfile to a Jekyll plugin to be aligned with other plugins and allows netlify to use it as well.

**Review by thaJeztah — APPROVED** [review]
Sitemap can be viewed here; https://deploy-preview-15250--docsdocker.netlify.app/sitemap.xml


LGTM!

**Comment by thaJeztah** [comment]
@usha-mandya PTAL

**Review by usha-mandya — APPROVED** [review]

**Pull request merged**